### DISCUSSION TITLE
Problem installing the library

### DISCUSSION THREAD

**Discussion by dead-security-ir** [TomSchimansky/CustomTkinter#2195] [General]
WARNING: Ignoring invalid distribution -llow (c:\users\hp 630\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution - (c:\users\hp 630\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -llow (c:\users\hp 630\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -0llow (c:\users\hp 630\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -1llow (c:\users\hp 630\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -2llow (c:\users\hp 630\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -illow (c:\users\hp 630\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (c:\users\hp 630\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', TimeoutError('_ssl.c:980: The handshake operation timed out'))': /simple/customtkinter/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', TimeoutError('_ssl.c:980: The handshake operation timed out'))': /simple/customtkinter/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', TimeoutError('_ssl.c:980: The handshake operation timed out'))': /simple/customtkinter/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', TimeoutError('_ssl.c:980: The handshake operation timed out'))': /simple/customtkinter/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', TimeoutError('_ssl.c:980: The handshake operation timed out'))': /simple/customtkinter/
ERROR: Could not find a version that satisfies the requirement customtkinter (from versions: none)
ERROR: No matching distribution found for customtkinter
WARNING: There was an error checking the latest version of pip.

**Comment by Akascape** [comment]
@dead-security-ir 

First of all, the first warning that you are getting  is due to some invalid package names. Check this solution: https://stackoverflow.com/questions/67442661/warning-ignoring-invalid-distribution-ip-c-python39-lib-site-packages-how-d


Secondly, the proxy warning are showing because of network issues. Check this solution: https://stackoverflow.com/a/47457000


Proper way to install package via pip:
- check your internet connection properly
- install latest version of python and pip
`python -m pip install --upgrade pip`
Minimum python version recommended: 3.9
- run the install command properly:
`python -m pip install customtkinter`
- then run this command:
`python -m pip install packaging`

If it still doesn't work, download the customtkinter folder manually and paste it in the project folder.

I hope it helps.

**Comment by Ezriral** [comment]
did not expect it to be so buggy. 2024 and still not able to install it via simple pip commands. hello 2012 : /
